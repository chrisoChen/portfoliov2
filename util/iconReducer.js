import GitHubIcon from "@mui/icons-material/GitHub";
import LinkedInIcon from "@mui/icons-material/LinkedIn";
import EmailIcon from "@mui/icons-material/Email";
import Link from "@mui/material/Link";

function iconReducer(links, color = "secondary.main") {
  let icon;

  const linkIcons = Object.keys(links).reduce((fragmentLinkIcons, currKey) => {
    switch (currKey) {
      case "github":
        icon = <GitHubIcon />;
        break;
      case "linkedin":
        icon = <LinkedInIcon />;
        break;
      case "email":
        icon = <EmailIcon />;
      default:
        icon = null;
        break;
    }

    fragmentLinkIcons[currKey] = (
      <Link
        href={links[currKey]}
        aria-label={`Link to my ${currKey}`}
        target="_blank"
        rel="noopener noreferrer"
        color={color}
        sx={{
          display: "flex",
          alignItems: "center",
          "&:hover": { opacity: "0.65", transition: "0.3s" },
        }}
      >
        {icon}
      </Link>
    );
    return fragmentLinkIcons;
  }, {});

  return linkIcons;
}

export default iconReducer;
