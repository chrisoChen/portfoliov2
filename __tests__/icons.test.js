import iconReducer from "../util/iconReducer";
import { render, screen } from "@testing-library/react";
import { v4 as uuidv4 } from "uuid";

let links;

beforeAll(() => {
  links = {
    github: "https://jestjs.io/docs/setup-teardown",
    linkedin: "https://developer.mozilla.org/en-US/",
  };
});

describe("Link reducer", () => {
  test("renders github logo icons", () => {
    const linkLogos = iconReducer(links);
    render(linkLogos.github);
    screen.getByTestId("GitHubIcon");
  });

  test("renders multiple logo icon links", () => {
    const linkLogos = iconReducer(links);
    const frag = render(
      Object.values(linkLogos).map((link) => {
        return <div key={uuidv4()}>{link}</div>;
      })
    );
    // screen.debug();
  });
});
