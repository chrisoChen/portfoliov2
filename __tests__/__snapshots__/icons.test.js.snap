// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link reducer renders multiple logo icon links 1`] = `undefined`;
